### PR TITLE
Add BiomeJS for Linting and Formatting JavaScript relates to #1295

### PR DESCRIPTION
**Pull request by oxdev03**
#### Summary:
- Added **BiomeJS** to lint and format JavaScript code.
- Integrated BiomeJS into the **devcontainer** and CI pipeline to ensure code consistency and quality.
- The CI now checks for linting issues in JavaScript files automatically.
- relates to #1295 

#### Why BiomeJS:
- https://github.com/biomejs/biome
- Biome offers a lighter and faster alternative to using ESLint and Prettier combined, providing the same functionality in a single tool.
- Simplifies configuration by reducing the need for multiple tools and configurations.

#### Open Points:
- Update the **Wiki** to include instructions on how to format and lint JS code using Biome.
- Check whether the  current lint rules are sufficient.

### CODE REVIEW

File: biome.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	},
+	"files": {
+		"ignoreUnknown": false,
+		"ignore": [],
+		"include": ["./app/javascript/**/*.js"]
+	},
+	"formatter": {
+		"enabled": true,
+		"indentStyle": "tab",
+		"useEditorconfig": true
```
Comment (zachgoll) [inline biome.json:16]: Currently, our `.editorconfig` has:

```
indent_style = space
indent_size = 2
```

Will this conflict since it's using tabs?

**Comment by zachgoll** [comment]
@oxdev03 thanks for the comprehensive integration here! Haven't used Biome yet, but seems like an overall good fit for the project regarding simplicity and performance.

Just jotting down the steps I took to get my local environment setup with VSCode for others:

1. Install Biome VSC extension
2. Add Biome as default formatter for JS in settings

```json
{
    "[javascript]": {
      "editor.formatOnSave": true,
      "editor.defaultFormatter": "biomejs.biome",
    }
}
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -52,6 +52,26 @@ jobs:
...
+        with:
+          node-version: "20"
+          cache: "npm"
+
+      - name: Install dependencies
+        run: npm install
+        shell: bash
+
+      - name: Lint/Format js code
+        run: npm run style:check
```
Comment (zachgoll) [inline .github/workflows/ci.yml:73]: To keep this as low-overhead as possible, I think we should just enforce the linting:

```
run: npm run lint
```

The formatting is a nice addition that I'll be using locally, but don't want that to be a requirement for new contributors

File: biome.json (hunk 1 above)
Comment (oxdev03) [reply biome.json:16]: good point, indentStyle overrides the imported editorconfig. I removed it, so its consistent across all files.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (oxdev03) [reply .github/workflows/ci.yml:73]: changed it, even if its not a best practice.

**Comment by oxdev03** [comment]
> ```json
> {
>     "[javascript]": {
>       "editor.formatOnSave": true,
>       "editor.defaultFormatter": "biomejs.biome",
>     }
> }
> ```

yes , the contributor would also need to have nodejs installed. I added the autosave to the vscode/settings.json file to the project.

If the DevContainer is used no additional steps are required.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (zachgoll) [reply .github/workflows/ci.yml]: @oxdev03 correct, not best practice and that's 100% fine at our stage/size

**Review by zachgoll — APPROVED** [review]
Looks good!  Thanks for setting this up!

**Pull request merged**